compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -9545,7 +9545,7 @@ private void handleSafeNavigation(BLangAccessExpression accessExpr, BType type,
 
         if (!(accessExpr.errorSafeNavigation || accessExpr.nilSafeNavigation)) {
             BType originalType = accessExpr.originalType;
-            if (TypeTags.isXMLTypeTag(originalType.tag)) {
+            if (TypeTags.isXMLTypeTag(originalType.tag) || isMapJson(originalType)) {
```

Comment:
The other types (object & records) supports field access expression don't need to be checked here. Because, the access keys are validated with their fields at compile time, so that it will never return a runtime error.